### PR TITLE
Move python tests to actions

### PR DESCRIPTION
**Pull request by benjeffery**
Travis recently capped the open source runners available, leading to long delays. Trying out actions as a replacement.

### CODE REVIEW

**Comment by AdminBot-tskit** [comment]
📖 Docs for this PR can be previewed [here](https://tskit-dev.github.io/tskit-docs/942/)

**Comment by AdminBot-tskit** [comment]
📖 Docs for this PR can be previewed [here](https://tskit-dev.github.io/tskit-docs/942/)

**Review by jeromekelleher — COMMENTED** [review]
Not sure it'll work out trying to share test code across the 3 platforms. I'd say it'd be simpler to split by windows,linux and osx

Particularly for htslib, this will never work on windows.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    branches: [main]
+
+jobs:
+  test:
+    name: Conda
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python: [ 3.6 ]
+        os:  [macos-latest, ubuntu-latest, windows-latest]
```
[the remark this answers is not in the document]
Comment (benjeffery) [reply .github/workflows/tests.yml:31]: I've had it working! Looks broke now as I'm getting conda caching working.

[the remark this answers is not in the document]
Comment (jeromekelleher) [reply .github/workflows/tests.yml:31]: Ahh. Good luck with that!

**Comment by jeromekelleher** [comment]
Also, just to confirm - is the idea we drop appveyor CI also?

**Comment by benjeffery** [comment]
> Also, just to confirm - is the idea we drop appveyor CI also?

I think so - will need you to deactivate it if we do as it seems to report in a failure with an absent config.

**Comment by jeromekelleher** [comment]
> I think so - will need you to deactivate it if we do as it seems to report in a failure with an absent config.

OK, just let me know when you want me to flick the switch

**Comment by benjeffery** [comment]
@jeromekelleher This is ready to go I think. Linux tests come back in 4-6min, windows in 6-7ish. Conda caching isn't working for OSX so it comes in at 11min. I might come back and fix that one day, but enough for now!

File: python/tests/test_fileobj.py [hunk 2]
```diff
@@ -283,7 +284,7 @@ def server_process(q):
     server.serve_forever()
 
 
-@pytest.mark.skipif(IS_WINDOWS, reason="Errors on Windows")
+@pytest.mark.skipif(IS_WINDOWS or IS_OSX, reason="Errors on Windows")
```
Comment (jeromekelleher) [inline python/tests/test_fileobj.py:287]: reason="Errors on bafflingly popular operating systems"?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      - name: Install conda deps
+        if: steps.cache.outputs.cache-hit != 'true'
+        shell: bash -l {0}
+        run: conda install --yes --file=python/requirements/CI-tests-conda/requirements.txt
+
+      - name: Install pip deps
+        if: steps.cache.outputs.cache-hit != 'true'
+        shell: bash -l {0}
+        env:
+          HTSLIB_CONFIGURE_OPTIONS: --disable-bz2
```
Comment (jeromekelleher) [inline .github/workflows/tests.yml:95]: It's not obvious to me why we're messing with htslib here - would it be better to try and get pysam from bioconda? (Presumably I tried this before and failed). It's a bit obscure anyway, a comment might help here.

**Review by jeromekelleher — COMMENTED** [review]
LGTM, minor comments

File: python/tests/test_fileobj.py (hunk 2 above)
Comment (benjeffery) [reply python/tests/test_fileobj.py]: Heh, I've added something more politically correct!

File: .github/workflows/tests.yml (hunk 3 above)
Comment (benjeffery) [reply .github/workflows/tests.yml]: You're right, wasn't needed.

**Comment by benjeffery** [comment]
@jeromekelleher I've fixed the OSX caching and the issues you found, should be good to merge and deactivate appveyor.

**Comment by jeromekelleher** [comment]
Awesome! I'll let you hit the merge button here. I think I've turned off the test running on Appveyor. Let's see how this github action works for a bit before we entirely delete the tskit project there (getting it set up was a PITA)

**Comment by jeromekelleher** [comment]
I've also disabled builds on Travis. Let's see how the actions go for a bit and start deleting the accounts in ~1 month if it's all going well.

**Comment by benjeffery** [comment]
Great, just rebased, will wait for green and merge. I expect a few teething issues, so yes lets keep old stuff around for a bit.

**Pull request merged**